core/src/impl/Kokkos_Core.cpp

```diff
@@ -304,21 +296,29 @@ void initialize_backends(const InitArguments& args) {
   Impl::ExecSpaceManager::get_instance().initialize_spaces(args);
 }
 
-void initialize_profiling(const InitArguments& args) {
-  Kokkos::Profiling::initialize(args.tool_lib);
-  if (args.tool_help) {
-    if (!Kokkos::Tools::printHelp(args.tool_args)) {
-      std::cerr << "Tool has not provided a help message" << std::endl;
-    }
+void initialize_profiling(const Tools::InitArguments& args) {
+  auto initialization_status =
+      Kokkos::Tools::Impl::initialize_tools_subsystem(args);
+  if (initialization_status.result ==
+      Kokkos::Tools::Impl::InitializationStatus::InitializationResult::
```

Comment:
Technically, you can directly write `Kokkos::Tools::Impl::InitializationStatus::success`.  Did you mean for `Kokkos::Tools::Impl::InitializationStatus::InitializationResult` to be a strong enumeration?

Reply:
No strong feelings about this one, game to spell it however